R/derive_param_tte.R

```diff
@@ -142,10 +159,259 @@
 #'
 #' @export
 #'
-#' @examples
+#' @examplesx
+#'
+#' @caption Add a basic time to event parameter
+#' @info For each subject the time to first adverse event should be created as a parameter.
+#'
+#' - The event source object is created using `event_source()` and the date is
+#'   set to adverse event start date.
+#' - The censor source object is created using `censor_source()` and the date is
+#'   set to end of study date.
+#' - Then these are passed to derive the time to event parameter with the provided
+#'   parameter descriptions (`PARAMCD` and `PARAM`).
+#' - Note the values of the censor variable (`CNSR`) that were derived here,
+#'   where the first subject has an event and the second does not.
+#' @code
 #' library(tibble)
 #' library(dplyr, warn.conflicts = FALSE)
-#' library(lubridate)
+#' library(lubridate, warn.conflicts = FALSE)
+#'
+#' adsl <- tribble(
+#'   ~USUBJID, ~TRTSDT,           ~EOSDT,            ~NEWDRGDT,
+#'   "01",     ymd("2020-12-06"), ymd("2021-03-06"), NA,
+#'   "02",     ymd("2021-01-16"), ymd("2021-02-03"), ymd("2021-01-03")
+#' ) %>%
+#'   mutate(STUDYID = "AB42")
+#'
+#' ae <- tribble(
+#'   ~USUBJID, ~AESTDT,           ~AESEQ, ~AEDECOD,
+#'   "01",     ymd("2021-01-03"),      1, "Flu",
+#'   "01",     ymd("2021-03-04"),      2, "Cough",
+#'   "01",     ymd("2021-03-05"),      3, "Cough"
+#' ) %>%
+#'   mutate(STUDYID = "AB42")
+#'
+#' ttae <- event_source(
+#'   dataset_name = "ae",
+#'   date = AESTDT,
+#'   set_values_to = exprs(
+#'     EVNTDESC = "AE",
+#'     SRCDOM = "AE",
+#'     SRCVAR = "AESTDTC",
+#'     SRCSEQ = AESEQ
+#'   )
+#' )
+#'
+#' eos <- censor_source(
+#'   dataset_name = "adsl",
+#'   date = EOSDT,
+#'   set_values_to = exprs(
+#'     EVNTDESC = "END OF STUDY",
+#'     SRCDOM = "ADSL",
+#'     SRCVAR = "EOSDT"
+#'   )
+#' )
+#'
+#' derive_param_tte(
+#'   dataset_adsl = adsl,
+#'   event_conditions = list(ttae),
+#'   censor_conditions = list(eos),
+#'   source_datasets = list(adsl = adsl, ae = ae),
+#'   set_values_to = exprs(
+#'     PARAMCD = "TTAE",
+#'     PARAM = "Time to First Adverse Event"
+#'   )
+#' ) %>%
+#'   select(USUBJID, STARTDT, PARAMCD, PARAM, ADT, CNSR, SRCSEQ)
+#'
+#' @caption Adding a by variable (`by_vars`)
+#' @info By variables can be added using the `by_vars` argument, e.g., now for
```

Comment:
@bundfussr maybe roclet 2.0 but be nice if you could hover over the arugment name and get a tooltip of the argument definition.

Reply:
maybe a discussion or running issue with all our ideas and dreams could be created??  i think some other more technical folks might want to get in on this!  thinking @StefanThoma here :)

but agree lets get to this CRAN and see if we get yelled at.